include/xen/public/grant_table.h

```diff
@@ -0,0 +1,397 @@
+
+/*
+ * Type of grant entry.
+ *  GTF_invalid: This grant entry grants no privileges.
+ *  GTF_permit_access: Allow @domid to map/access @frame.
+ *  GTF_accept_transfer: Allow @domid to transfer ownership of one page frame
+ *                       to this guest. Xen writes the page number to @frame.
+ *  GTF_transitive: Allow @domid to transitively access a subrange of
+ *                  @trans_grant in @trans_domid.  No mappings are allowed.
+ */
+#define GTF_invalid				(0U << 0)
+#define GTF_permit_access			(1U << 0)
+#define GTF_accept_transfer			(2U << 0)
+#define GTF_transitive				(3U << 0)
+#define GTF_type_mask				(3U << 0)
+
+/*
+ * Subflags for GTF_permit_access and GTF_transitive.
+ *  GTF_readonly: Restrict @domid to read-only mappings and accesses. [GST]
+ *  GTF_reading: Grant entry is currently mapped for reading by @domid. [XEN]
+ *  GTF_writing: Grant entry is currently mapped for writing by @domid. [XEN]
+ * Further subflags for GTF_permit_access only.
+ *  GTF_PAT, GTF_PWT, GTF_PCD: (x86) cache attribute flags to be used for
+ *                             mappings of the grant [GST]
+ *  GTF_sub_page: Grant access to only a subrange of the page.  @domid
+ *                will only be allowed to copy from the grant, and not
+ *                map it. [GST]
+ */
+#define _GTF_readonly				(2)
+#define GTF_readonly				(1U << _GTF_readonly)
+#define _GTF_reading				(3)
+#define GTF_reading				(1U << _GTF_reading)
+#define _GTF_writing				(4)
+#define GTF_writing				(1U << _GTF_writing)
+#define _GTF_PWT				(5)
+#define GTF_PWT					(1U << _GTF_PWT)
+#define _GTF_PCD				(6)
+#define GTF_PCD					(1U << _GTF_PCD)
+#define _GTF_PAT				(7)
+#define GTF_PAT					(1U << _GTF_PAT)
+#define _GTF_sub_page				(8)
+#define GTF_sub_page				(1U << _GTF_sub_page)
+
+/*
+ * Subflags for GTF_accept_transfer:
+ *  GTF_transfer_committed: Xen sets this flag to indicate that it is committed
+ *      to transferring ownership of a page frame. When a guest sees this flag
+ *      it must /not/ modify the grant entry until GTF_transfer_completed is
+ *      set by Xen.
+ *  GTF_transfer_completed: It is safe for the guest to spin-wait on this flag
+ *      after reading GTF_transfer_committed. Xen will always write the frame
+ *      address, followed by ORing this flag, in a timely manner.
+ */
+#define _GTF_transfer_committed			(2)
+#define GTF_transfer_committed			(1U << _GTF_transfer_committed)
+#define _GTF_transfer_completed			(3)
+#define GTF_transfer_completed			(1U << _GTF_transfer_completed)
+
+/***********************************
+ * GRANT TABLE QUERIES AND USES
+ */
+
+/* ` enum neg_errnoval
+ * ` HYPERVISOR_grant_table_op(enum grant_table_op cmd,
+ * `                           void *args,
+ * `                           unsigned int count)
+ * `
+ *
+ * @args points to an array of a per-command data structure. The array
+ * has @count members
+ */
+
+/* ` enum grant_table_op { // GNTTABOP_* => struct gnttab_* */
```

Comment:
Hi, I saw there are several '`'s, and it seems useless code which has been toggled to comments. Could we delete them?

Reply:
As I understand this is original explanation for grant table ops parameters and retval in Xen public headers, I prefer to left it as is due to same reasons as __XEN_INTERFACE_VERSION__  discussion above.